File: servers/python/v1/store.py
class BaseRpcException(Exception):
  def __init__(self, value):
    self.value = value

  def __str__(self):
    return repr(self.value)

class UserDataInvalid(BaseRpcException):
  pass

class RecordNotFound(BaseRpcException):
  pass

class MaxTodosExceeded(BaseRpcException):
  pass

class Store(object):

  def __init__(self):
    self.cache = {}
    self.next_id = 0

  def save(self, data):
    self.next_id = self.next_id + 1

    todo = {
      'id' : self.next_id,
      'title' : data['title'],
      'completed' : data['completed']
    }

    self.cache[self.next_id] = todo

    return todo

  def get_all(self):
    return self.cache.values()

  def update(self, id, data):
    if id not in self['cache']:
      raise RecordNotFound("No record found with id" + id)

    todo = {
      'id' : id,
      'title' : data['title'],
      'completed' : data['completed']
    }

    self['cache'][id] = todo

    return todo

  def delete(self, id):
    return (not not self.cache.pop(id))

File: servers/golang/store/store.go
package store

import t "../todos"

type Store struct {
  todos map[int64]t.Todo
  next_id int64
}

func NewStore() *Store {
  return &Store{make(map[int64]t.Todo),0}
}

func (s Store) GetAll() ([]t.Todo, error) {
  v := make([]t.Todo, 0, len(s.todos))

  for  _, value := range s.todos {
    v = append(v, value)
  }

  return v, nil
}

func (s Store) Save(properties t.TodoProperties) (t.Todo, error) {
  todo := t.Todo{properties,1}
  s.todos[todo.Id] = todo
  return todo, nil
}

func (s Store) Update(todo t.Todo) (t.Todo, error) {
  for cached_id := range s.todos {
    if todo.Id == cached_id {
      s.todos[cached_id] = todo
    }
  }

  return todo, nil
}

func (s Store) Delete(id int64) (bool, error) {
  for cached_id := range s.todos {
    if id == cached_id {
      delete(s.todos,id)
    }
  }

  return true, nil
}
